tests/core/objects/switch_base_test.py

```diff
@@ -96,3 +96,11 @@ def test_ending_transaction_failed_leaves_the_flag_on_and_raises(self):
             self.switch.end_transaction()
 
         assert_that(self.switch.in_transaction, is_(True))
+
+    def test_not_implemented_methods_raise_with_their_names(self):
+        so = SwitchOperations()
+        for method_name in SwitchOperations.__dict__:
```

Comment:
dir(SwitchOperations)

Reply:
The problem with dir() is that it will return all the super class methods as well. We don't what that.